src/mesh/mesh.jl

```diff
@@ -66,8 +76,13 @@ function generate_mesh()
   periodicity = parameter("periodicity", true)
 
   # Create mesh
-  @timeit timer() "creation" mesh = TreeMesh(Val{ndims_}(), n_cells_max, domain_center,
-                                             domain_length, periodicity)
+  if is_parallel()
+    @timeit timer() "creation" mesh = TreeMesh(ParallelTree{ndims_}, n_cells_max,
+                                               domain_center, domain_length, periodicity)
+  else
+    @timeit timer() "creation" mesh = TreeMesh(Tree{ndims_}, n_cells_max, domain_center,
+                                               domain_length, periodicity)
+  end
```

Comment:
What about trying to reduce the amount of duplicated code, e.g.

```suggestion
  if is_parallel()
    tree_type = ParallelTree{ndims_}
  else
    tree_type = Tree{ndims_}
  end
  @timeit timer() "creation" mesh = TreeMesh(tree_type, n_cells_max, domain_center,
                                             domain_length, periodicity)
```

Another question: Should we rename `Tree` to `SerialTree` to make it more similar to `ParallelTree`?

Reply:
@ranocha In cc29ac4 I renamed `Tree` -> `SerialTree`. I also refactored the code to reduce code duplication, thus all tree functions are now defined on the `AbstractTree` in `abstract_tree.jl` (with default implementations for most methods), while `serial_tree.jl` and `parallel_tree.jl` contain only the serial/parallel specific stuff. Could you please re-review and let me know what you think in terms of Julianess and if this separation makes sense in general?